src/string_search.h

```diff
@@ -274,111 +229,133 @@ inline uint8_t GetHighestValueByte(uint16_t character) {
 inline uint8_t GetHighestValueByte(uint8_t character) { return character; }
 
 
-template <typename PatternChar, typename SubjectChar>
-inline size_t FindFirstCharacter(Vector<const PatternChar> pattern,
-                              Vector<const SubjectChar> subject, size_t index) {
-  const PatternChar pattern_first_char = pattern[0];
+// Searches for a byte value in a memory buffer, back to front
+// Uses memrchr(3) on systems which support it, for speed
+// Falls back to a vanilla for loop on systems which don't
+inline const void* MemrchrFill(const void* haystack, uint8_t needle,
+                               size_t haystack_len) {
+#ifdef _GNU_SOURCE
+  return memrchr(haystack, needle, haystack_len);
+#else
+  const uint8_t *haystack8 = reinterpret_cast<const uint8_t*>(haystack);
```

Comment:
Sure, I'll change it. Just noting I've always seen the style guides and linters enforcing the opposite -- star goes to the right. That's because:

**a, b, and c are all `int`s**
```
int a, b, c;
```


**a is an `int*`, b and c are `int`s**
```
int* a, b, c;
```

**a, b, and c are all `int*`**
```
int *a, *b, *c;
```